### PR TITLE
7388 update skill showcase

### LINKED ISSUE #7388: ✨ "Update a skill showcase" pages
image

## ✨ Feature

Allows a user to define a Skill Showcase list.

## 🕵️ Details

On this page, users can
- add new skills (up to a max, which differs for the different lists)
- remove skills
- reorder items (order will be saved)
- save or quit

The list should use the Repeater component we built recently.

**Note**
The `Repeater` component  is wrapped in a `fieldset`, create a prop that allows to render a `div` instead for using forms.

QUESTION: how do we handle the 4 different versions of this page? They can definitely share a component at least, with just a few props differing between them. But does each have its own url? Is it one url with url-parameters to differentiate them?

QUESTION: what are we naming this url(s)?
ANSWERS:
- `/applicant/profile-and-applications/skills/showcase/top-5-behavioural-skills`
- `/applicant/profile-and-applications/skills/showcase/top-10-technical-skills`
- `/applicant/profile-and-applications/skills/showcase/3-behavioural-skills-to-improve`
- `/applicant/profile-and-applications/skills/showcase/5-technical-skills-to-train`

## 📸 Screenshots

For convenience only - remember to work from figma to stay up-to-date!

<details><summary>Top 5 behavioural skills (Empty)</summary>

</details>

<details><summary>Top 5 behavioural skills (Complete)</summary>

</details>

<details><summary>Top 10 technical skills (Empty)</summary>

</details>

<details><summary>Top 10 technical skills (Complete)</summary>

</details>

<details><summary>3 behavioural skills you'd like to improve (Empty)</summary>

</details>

<details><summary>3 behavioural skills you'd like to improve (Complete)</summary>

</details>

<details><summary>5 technical skills you'd like training in (Empty)</summary>

</details>

<details><summary>5 technical skills you'd like training in (Complete)</summary>

</details>

<details><summary>Add a skill dialog</summary>

</details>

## 🌎 Localization

(optional) Provide any new copy along with translations available.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the feature was properly implemented.

- [ ] This is behind SKILL_LIBRARY feature flag
- [ ] Page(s) exist at new route(s)
- [ ] Has a table of contents, with one item
- [ ] Shows existing list, in correct order
- [ ] Uses our existing repeater component for accessible interactions
- [ ] Allows editing and saving of list 
- [ ] Cancel and go back is a simple link
- [ ] Ensure the repeater element is adjusted, if necessary, to match the design (likely white space changes)
- [ ] Save and return does what it says
- [ ] Ensure that the changes done in #7394 propagate here, and ensure that the `My library` prop is enabled for these pages
- [ ] If a user adds a skill to their showcase that isn't in their library, add it to their library

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #7388 

## 👋 Introduction

- Allows user to update/create a new UserSkill to showcase.
- Allows applicant to create a list of skills (top / to improve), and select the order of importance using the Repeater component.
- Adds 4 new pages to applicant side
  - `/applicant/profile-and-applications/skills/showcase/top-5-behavioural-skills`
  - `/applicant/profile-and-applications/skills/showcase/top-10-technical-skills`
  - `/applicant/profile-and-applications/skills/showcase/3-behavioural-skills-to-improve`
  - `/applicant/profile-and-applications/skills/showcase/5-technical-skills-to-train`

## 🕵️ Details

- Go through AC and ensure nothing was missed #7388 

## 🧪 Testing

1. Login as applicant/admin/etc.
2. Go to each of the new pages stated above (you can find a link to each page on the [skill-showcase page](http://localhost:8000/en/applicant/profile-and-applications/skills/showcase)
3. Ensure all AC is correct, ensure all actions work correctly (creating/updating/rearranging order), ensure all styling/copy is correct.

### CODE REVIEW

File: apps/web/src/pages/Skills/components/UpdateSkillShowcase.tsx [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+                                data-h2-display="base(flex)"
+                                data-h2-justify-content="base(space-between)"
+                              >
+                                <span>
+                                  {getLocalizedName(
+                                    getSkill(item.skill)?.name,
+                                    intl,
+                                  )}
+                                </span>
+                                <span>
```
Comment (esizer) [inline apps/web/src/pages/Skills/components/UpdateSkillShowcase.tsx:318]: Looks like this should have a font-weight of 400 and a colour of `black.light`

File: apps/web/src/pages/Skills/SkillShowcasePage.tsx [hunk 2]
```diff
@@ -217,7 +217,7 @@ export const SkillShowcase = ({
                     description: "Description of a users top technical skills",
                   })}
                   editLink={{
-                    href: paths.skillShowcase(), // TO DO: Update with proper path in #7388
+                    href: paths.topTechnicalSkills(), // TO DO: Update with proper path in #7388
```
Comment (esizer) [inline apps/web/src/pages/Skills/SkillShowcasePage.tsx:220]: Thanks for adding these, looks like the comment can be removed now 👍

**Review by esizer — CHANGES_REQUESTED** [review]
Wow, this is awesome! ⭐  I noticed a few minor things along with some questions.

File: apps/web/src/pages/Skills/SkillShowcasePage.tsx (hunk 2 above)
Comment (yonikid15) [reply apps/web/src/pages/Skills/SkillShowcasePage.tsx]: Oops, fixed by [6abfbfe](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7933/commits/6abfbfe693587a03f54f426e7c9a3c8d1ebc7b36)

File: apps/web/src/pages/Skills/ImproveBehaviouralSkillsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      userSkills
+        ?.filter(
+          (userSkill) =>
+            userSkill.skill.category === SkillCategory.Behavioural &&
+            userSkill.improveSkillsRank,
+        )
+        .sort(
+          (a, b) =>
+            (a.improveSkillsRank ?? 0) > (b.improveSkillsRank ?? 0) ? 1 : -1, // sort initial user skills
+        )
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/pages/Skills/ImproveBehaviouralSkillsPage.tsx:180]: They are not for some reason. I quickly made that sort function, but I'll take a look as to why it's unsorted. My guess is it's returning it in order of created_at.

File: apps/web/src/pages/Skills/ImproveBehaviouralSkillsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  const pageDescription = intl.formatMessage({
+    defaultMessage:
+      "Help managers understand your potential behavioural skill learning opportunities.",
+    description: "Page description for the improve behavioural skills page",
+    id: "EBE5Gv",
+  });
+
+  const pageBlurb = intl.formatMessage({
+    defaultMessage:
+      "Help managers understand where you’d like to improve behavioural skills. You can edit this showcase at any time and are free to order the skills however you like. Skills that you add to the showcase that aren’t already in your library will be added for you automatically. ",
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/pages/Skills/ImproveBehaviouralSkillsPage.tsx]: Nice catch, fixed by [6b75a0a](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7933/commits/6b75a0a70179831a3d52e16009c836837ab35ef3)

File: apps/web/src/pages/Skills/components/UpdateSkillShowcase.tsx (hunk 1 above)
Comment (yonikid15) [reply apps/web/src/pages/Skills/components/UpdateSkillShowcase.tsx]: Thanks! Fixed by [b66f07f](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7933/commits/b66f07fc8078d74e92fd972ac6626f6ff8cb9100)

File: apps/web/src/pages/Skills/components/UpdateSkillShowcase.tsx [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+                                    ? intl.formatMessage(
+                                        item.category ===
+                                          SkillCategory.Behavioural
+                                          ? getBehaviouralSkillLevel(
+                                              item.skillLevel,
+                                            )
+                                          : getTechnicalSkillLevel(
+                                              item.skillLevel,
+                                            ),
+                                      )
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply apps/web/src/pages/Skills/components/UpdateSkillShowcase.tsx:332]: Oh wow thanks for spotting this, fixed by [457eb53](https://github.com/GCTC-NTGC/gc-digital-talent/pull/7933/commits/457eb53b06963a5f2f810f4a544b1712d44d81d5)

**Review by esizer — APPROVED** [review]
Great work! Everything looks good to me! 👍 

One thing I thought when testing this was, "I'd really enjoy an edit button for the skills within the repeater to edit the skill level, etc in place". That is not part of the scope, something I just think we could easily do by putting the edit form in a dialog here 🤷‍♀️

**Comment by tristan-orourke** [comment]
> One thing I thought when testing this was, "I'd really enjoy an edit button for the skills within the repeater to edit the skill level, etc in place". That is not part of the scope, something I just think we could easily do by putting the edit form in a dialog here 🤷‍♀️

Good idea! Want to submit a ticket for designer consideration? https://github.com/GCTC-NTGC/design-gc-digital-talent/issues/new?assignees=&labels=&projects=&template=usability-improvement-template.md&title=

**Pull request merged**